### PR TITLE
run tests on wasm32-unknown-unknown

### LINKED ISSUE #79: run tests on wasm32-unknown-unknown

### PR DESCRIPTION
**Pull request by gnzlbg**
Closes #79

### CODE REVIEW

**Comment by gnzlbg** [comment]
@alexcrichton this runs the full testsuite on wasm32-unknown-unknown except for the `#[should_panic]` tests which I have currently disabled.

**Comment by gnzlbg** [comment]
Having to use the `#[wasm_bindgen_macro]`, `pub mod`, `pub fn`, etc. for tests, disable `#[should_panic]`, are the only "issues" i've encountered. 

_I think_ we could have a tiny compiler `feature(plugin)` that replaces `#[test]` with the appropriate stuff, and that it would be enough to include that for the wasm target in the top-level `Cargo.toml`. 

That would make testing Rust crates for `wasm` a two-liner change on nightly, instead of something that one has to take care of for every test.

**Comment by gnzlbg** [comment]
@alexcrichton 

```
{ Error: Cannot find module 'env'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/checkout/target/wasm32-unknown-unknown/wbg-tmp/wasm-bindgen-test_bg.js:6:18)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3) code: 'MODULE_NOT_FOUND' }
```

This happens in debug builds. I am trying to run the tests in release mode to see if the issue persists.

File: src/api/slice/write_to_slice.rs [hunk 1]
```diff
@@ -76,12 +76,12 @@ macro_rules! impl_slice_write_to_slice {
...
             interpolate_idents! {
-                mod [$id _slice_write_to_slice] {
+                pub mod [$id _slice_write_to_slice] {
                     use super::*;
                     use iter::Iterator;
 
-                    #[test]
-                    fn write_to_slice_unaligned() {
+                    #[cfg_attr(not(target_arch = "wasm32"), test)] #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+                    pub fn write_to_slice_unaligned() {
```
Comment (alexcrichton) [inline src/api/slice/write_to_slice.rs]: While the modules need to be publich (a [known bug](https://github.com/rustwasm/wasm-bindgen/issues/201)) you shouldn't need to use `pub` on the methods themselves

**Comment by alexcrichton** [comment]
Hm that basically means that there was an unresolved symbol, do you know if the test expect a symbol from libc or a C library or something like that?

**Comment by gnzlbg** [comment]
> Hm that basically means that there was an unresolved symbol, do you know if the test expect a symbol from libc or a C library or something like that?

It fails only in debug mode; in release everything works as expected.

**Pull request merged**